### PR TITLE
XdgMimeApps: Fix a memory leak

### PR DESCRIPTION
**Pull request by luis-pereira**

### CODE REVIEW

File: src/qtxdg/xdgmimeapps.cpp [hunk 1]
```diff
@@ -96,6 +96,7 @@ QList<XdgDesktopFile *> XdgMimeApps::categoryApps(const QString &category)
         if (!categories.isEmpty() && (categories.contains(cat) || categories.contains(QL1S("X-") + cat)))
             dl.append(df);
     }
+    qDeleteAll(apps);
```
Comment (tsujan) [inline src/qtxdg/xdgmimeapps.cpp:99]: Doesn't this make `df` consist of dangling pointers? I think we may need to delete only those pointers that aren't included in `df`.

Comment (luis-pereira) [reply src/qtxdg/xdgmimeapps.cpp]: You were faster. That was a debug branch to find the source of the leak. Pushed the wrong branch. Fixed now.

**Comment by luis-pereira** [comment]
@tsujan ping.

**Comment by tsujan** [comment]
Oh, it's very good.

**Review by tsujan — APPROVED** [review]
GTM

**Pull request merged**